Source/Core/Core/Core.cpp

```diff
@@ -146,7 +154,7 @@ bool IsRunningInCurrentThread()
 
 bool IsCPUThread()
 {
-	return (s_cpu_thread.joinable() ? (s_cpu_thread.get_id() == std::this_thread::get_id()) : !s_is_started);
+	return s_cpu_thread.joinable() ? tls_is_cpu_thread : !s_is_started;
```

Comment:
The "problem" is that the way HLE BS2 works is a little weird: we actually start running the emulation before the CPU thread starts, in CBoot::BootUp(), using a miniature interpreter loop, so the startup thread is actually the CPU thread for a short period of time.